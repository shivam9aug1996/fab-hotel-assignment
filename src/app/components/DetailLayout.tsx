import Back from "./Back";

const DetailLayout = ({
  children,
  title,
}: {
  children: React.ReactNode;
  title: string;
}) => (
  <div>
    <Back />
    <div className="p-8 max-w-2xl mx-auto bg-white shadow-2xl rounded-lg border border-gray-200">
      <div className="relative mb-6">
        <div className="absolute inset-0 bg-gradient-to-r from-blue-100 to-purple-100 rounded-lg opacity-50"></div>
        <h1 className="relative text-4xl font-extrabold mb-4 text-gray-900 p-5">
          {title}
        </h1>
      </div>
      <div className="space-y-6">{children}</div>
    </div>
  </div>
);

export default DetailLayout;
