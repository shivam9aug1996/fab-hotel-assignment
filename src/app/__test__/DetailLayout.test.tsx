import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import DetailLayout from "../components/DetailLayout";

jest.mock("next/navigation", () => ({
  useRouter: () => ({
    back: jest.fn(),
  }),
}));

describe("DetailLayout", () => {
  it("renders DetailLayout", () => {
    const { container } = render(
      <DetailLayout title="Test Title">{<></>}</DetailLayout>
    );
    expect(container).toMatchSnapshot();
  });
});
