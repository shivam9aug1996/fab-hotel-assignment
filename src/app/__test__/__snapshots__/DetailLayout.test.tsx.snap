// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailLayout renders DetailLayout 1`] = `
<div>
  <div>
    <button
      class="bg-gray-800 text-white py-2 px-4 rounded mt-6 absolute top-0 left-6"
    >
      Back
    </button>
    <div
      class="p-8 max-w-2xl mx-auto bg-white shadow-2xl rounded-lg border border-gray-200"
    >
      <div
        class="relative mb-6"
      >
        <div
          class="absolute inset-0 bg-gradient-to-r from-blue-100 to-purple-100 rounded-lg opacity-50"
        />
        <h1
          class="relative text-4xl font-extrabold mb-4 text-gray-900 p-5"
        >
          Test Title
        </h1>
      </div>
      <div
        class="space-y-6"
      />
    </div>
  </div>
</div>
`;
